src/model/MultiValueColumn.ts

```diff
@@ -0,0 +1,257 @@
+  }
+
+  q3() {
+    return (d3.quantile(this.aVal.sort(numSort), 0.75)) - (d3.quantile(this.bVal.sort(numSort), 0.75));
+  }
+
+}
+
+
+export function numSort(a, b) {
+
+  return a - b;
+}
+
+enum Sort {
+
+  min, max, median, q1, q3, mean
+}
+
+export interface IMultiValueColumn {
+  isLoaded(): boolean;
+  getNumber(row: any, index: number): number[];
+}
+
+export interface IMultiValueColumnDesc extends IValueColumnDesc <number[]> {
+
+  readonly domain?: number [];
+  readonly sort?: string;
+  readonly threshold?: number;
+  readonly dataLength?: number;
+  readonly colorRange?: any;
+  readonly min?: number;
+  readonly max?: number;
+
+}
+
+
+// interface IMultiValueColumnDesc extends IDataStat {
+//
+//   readonly threshold: number,
+//   readonly dataLength: number,
+//   readonly colorRange: any
+// }
+
+
+export default class MultiValueColumn extends ValueColumn<number[]> implements IBoxPlotColumn,IMultiValueColumn {
+
+  private userSort;
+  private min;
+  private max;
+  private sort;
+  private threshold;
+  private dataLength;
+  private colorRange;
+
+  private data;
+  private colorScale: d3.scale.Linear<number, string> = d3.scale.linear<number, string>();
+  private xposScale: d3.scale.Linear<number, number> = d3.scale.linear();
+  private yposScale: d3.scale.Linear<number, number> = d3.scale.linear();
+  private verticalBarScale: d3.scale.Linear<number,number> = d3.scale.linear();
+
+
+  constructor(id: string, desc: IMultiValueColumnDesc) {
+    super(id, desc);
+    this.min = d3.min(desc.domain);
+    this.max = d3.max(desc.domain);
+    this.dataLength = desc.dataLength;
+    this.threshold = desc.threshold || 0;
+    this.colorRange = desc.colorRange || ['blue', 'red'];
+    this.sort = desc.sort || 'min';
+
+    this.data = {
+      min: this.min,
+      max: this.max,
+      dataLength: this.dataLength,
+      threshold: this.threshold,
+      sort: this.sort,
+      colorRange: this.colorRange
+    };
+
+    this.userSort = this.sort;
+
+    const rendererList = [{type: 'multiValueHeatmap', label: 'Heatmap'},
+      {type: 'boxplot', label: 'Boxplot'},
+      {type: 'sparkline', label: 'Sparkline'},
+      {type: 'threshold', label: 'Threshold'},
+      {type: 'verticalbar', label: 'VerticalBar'}];
+
+    this.setRendererList(rendererList);
+
+    this.defineColorRange();
+
+  }
+
+
+  checkColorValues(): String[] {
+    if (this.data.colorRange.length > 2) {
+      const minColor = this.data.colorRange[0];
+      const zeroColor = this.data.colorRange[1];
+      const maxColor = this.data.colorRange[2];
+      const colorRange = [minColor, zeroColor, maxColor];
+      return colorRange;
+
+    } else {
+      const minColor = this.data.colorRange[0];
+      const zeroColor = 'white';
+      const maxColor = this.data.colorRange[1];
+      const colorRange = [minColor, zeroColor, maxColor];
+      return colorRange;
+    }
+  }
+
+
+  private defineColorRange() {
+    const colorValues: any = this.checkColorValues();
```

Comment:
when creating the scale you specify the type of type of the scaling using generics. In this case, it seems you declared it having a number domain instead of a string domain